### PR TITLE
    Using IPC::Open3 to capture STDERR

### PR DESCRIPTION
**Pull request by wjn740**

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 1]
```diff
@@ -353,16 +354,47 @@ sub _commit_git {
     }
     my @git = ('git', '--git-dir', "$dir/.git", '--work-tree', $dir);
     my @files = ($dir . '/' . $name . '.json', $dir . '/' . $name . '.png');
-    if (system(@git, 'add', @files) != 0) {
+    my ($child_in, $child_out, $child_err);
+    eval { $pid = open3($child_in, $child_out, $child_err, @git, 'add', @files); };
```
Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/Step.pm]: Your $pid is not declared

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 2]
```diff
@@ -353,16 +354,47 @@ sub _commit_git {
...
+    die "failed to git add $name";
+    waitpid($pid, 0);
+    my $ret = $? >> 8;
+
+    while (my $output = <$child_err>) {
+        chomp "$output";
```
Comment (aaannz) [inline lib/OpenQA/WebAPI/Controller/Step.pm]: this won't work, use it without ""
```perl
chomp $output;
```

**Comment by aaannz** [comment]
But still, this issue is already almost fixed.
And you have some other problems within the code (like no checks for $@ after eval and dying everytime). I suggest to close this PR and rather focus on some other available tasks from https://progress.opensuse.org/projects/openqav3/issues

**Comment by coolo** [comment]
I agree

**Comment by wjn740** [comment]
Thank you sir. I got it.

**Pull request closed without merging**